pkg/daemons/control/server.go

```diff
@@ -107,6 +114,42 @@ func Server(ctx context.Context, cfg *config.Control) error {
 	return nil
 }
 
+func waitForAPIServer(cfg *config.Control) error {
+	f, err := os.Open(cfg.Runtime.KubeConfigAdmin)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	b, err := ioutil.ReadAll(f)
+	if err != nil {
+		return err
+	}
+
+	restconfig, err := clientcmd.RESTConfigFromKubeConfig(b)
+	if err != nil {
+		return err
+	}
+
+	clientset, err := clientset.NewForConfig(restconfig)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < 60; i++ {
+		info, err := clientset.Discovery().ServerVersion()
+		if err != nil {
+			logrus.Warnf("apiserver is not yet up and running, will wait one second until next check")
+			time.Sleep(1 * time.Second)
+		} else {
+			logrus.Infof("apiserver %s is up and running", info)
+			break
+		}
+	}
+
+	return nil
+}
```

Comment:
What if we've been waiting 60 seconds, shouldn't it error out then? Now it seems to just continue as if the api server was up and running even though we haven't gotten a response? Did I miss something?

Reply:
You're right, I'll fix that. I'm not sure this is the best way to solve this.

Reply:
Now the code always fail as even if we enter the else statement we `return err` at the end.
If we do `return nil` in the else statement it shouldn't return an error if everything is good.

Reply:
`err` at the end is `nil` after a successful call to the `apiserver` but maybe that's not clear enough. Better just to return `nil` in the `else`. I'll fix

Reply:
Aah, missed that kinda used to early out code and don't work with go much at all :p